packages/block-library/src/button/style.scss

```diff
@@ -36,6 +36,10 @@ $blocks-button__height: 56px;
 	}
 }
 
+.wp-gs .wp-block-button__link:not(.has-background) {
+	background-color: var(--wp-block-core-button--color--background, var(--wp-color--primary, $dark-gray-700));
```

Comment:
@jorgefilipecosta Halloo!!! Just wanted to verify! It looks like there are two hyphens after the prefix, example:

```
--wp-block-core-button--color--background
```

Note: `--color--background`

Is this intentional?

Reply:
Yes, it is intentional to remove ambiguity. If we don't add two hyphens, we don't know if the variable represents this object { ... { color: { background: ... } } } or this one { ...{ 'color-background': ... } }. Having two different object shapes generate the same variables is something that would, for sure, cause problems as an object may overwrite the styles of another object when it is not supposed.

Reply:
Cool! I'm down with this approach 👍 
Thank you!